### DISCUSSION TITLE
Error in running slurm web on localhost

### DISCUSSION THREAD

**Discussion by Talavig** [rackslab/Slurm-web#315] [Q&A] [answered]
I try to run slurm-web on a single, localhost cluster as a test. when i try to go to localhost:5011 from my browser, i get the following message:
`Server error: The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application.`
when i look at the logs for slurm-web-agen, i see the following:
`Thread-858 (process_request_thread): [INFO] 127.0.0.1 - - [14/Jul/2024 13:58:32] "GET /v3.1.0/stats HTTP/1.1" 500 -
Jul 14 13:58:32 standalone slurm-web-agent[7637]: simplejson.errors.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
Jul 14 13:58:32 standalone slurm-web-agent[7637]:     return self.scan_once(s, idx=_w(s, idx).end())
Jul 14 13:58:32 standalone slurm-web-agent[7637]:   File "/usr/lib/python3/dist-packages/simplejson/decoder.py", line 400, in raw_decode
Jul 14 13:58:32 standalone slurm-web-agent[7637]:     obj, end = self.raw_decode(s)
Jul 14 13:58:32 standalone slurm-web-agent[7637]:   File "/usr/lib/python3/dist-packages/simplejson/decoder.py", line 370, in decode
Jul 14 13:58:32 standalone slurm-web-agent[7637]:     return _default_decoder.decode(s)
Jul 14 13:58:32 standalone slurm-web-agent[7637]:   File "/usr/lib/python3/dist-packages/simplejson/__init__.py", line 525, in loads
Jul 14 13:58:32 standalone slurm-web-agent[7637]:     return complexjson.loads(self.text, **kwargs)
Jul 14 13:58:32 standalone slurm-web-agent[7637]:   File "/usr/lib/python3/dist-packages/requests/models.py", line 900, in json
Jul 14 13:58:32 standalone slurm-web-agent[7637]:     result = response.json()
Jul 14 13:58:32 standalone slurm-web-agent[7637]:   File "/usr/lib/python3/dist-packages/slurmweb/views/agent.py", line 51, in slurmrest
Jul 14 13:58:32 standalone slurm-web-agent[7637]:     items = func(*args)
Jul 14 13:58:32 standalone slurm-web-agent[7637]:   File "/usr/lib/python3/dist-packages/slurmweb/views/agent.py", line 76, in filter_fields
Jul 14 13:58:32 standalone slurm-web-agent[7637]:     return func(*args)
Jul 14 13:58:32 standalone slurm-web-agent[7637]:   File "/usr/lib/python3/dist-packages/slurmweb/views/agent.py", line 88, in _cached_data
Jul 14 13:58:32 standalone slurm-web-agent[7637]:     return _cached_data(
Jul 14 13:58:32 standalone slurm-web-agent[7637]:   File "/usr/lib/python3/dist-packages/slurmweb/views/agent.py", line 101, in _cached_jobs
Jul 14 13:58:32 standalone slurm-web-agent[7637]:     for job in _cached_jobs():
Jul 14 13:58:32 standalone slurm-web-agent[7637]:   File "/usr/lib/python3/dist-packages/slurmweb/views/agent.py", line 236, in stats
Jul 14 13:58:32 standalone slurm-web-agent[7637]:     return view(*args, **kwargs)
Jul 14 13:58:32 standalone slurm-web-agent[7637]:   File "/usr/lib/python3/dist-packages/rfl/web/tokens.py", line 93, in wrapped
Jul 14 13:58:32 standalone slurm-web-agent[7637]:     return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
Jul 14 13:58:32 standalone slurm-web-agent[7637]:   File "/usr/lib/python3/dist-packages/flask/app.py", line 1499, in dispatch_request
Jul 14 13:58:32 standalone slurm-web-agent[7637]:     rv = self.dispatch_request()
Jul 14 13:58:32 standalone slurm-web-agent[7637]:   File "/usr/lib/python3/dist-packages/flask/app.py", line 1513, in full_dispatch_request
Jul 14 13:58:32 standalone slurm-web-agent[7637]:     rv = self.handle_user_exception(e)
Jul 14 13:58:32 standalone slurm-web-agent[7637]:   File "/usr/lib/python3/dist-packages/flask/app.py", line 1515, in full_dispatch_request
Jul 14 13:58:32 standalone slurm-web-agent[7637]:     response = self.full_dispatch_request()
Jul 14 13:58:32 standalone slurm-web-agent[7637]:   File "/usr/lib/python3/dist-packages/flask/app.py", line 2070, in wsgi_app`
Its important to mention that i only manage to reach slurmrestd by using sudo in my curl command. maybe that's related?
Why is this error thrown? 
In addition, can you only reach slurmrestd through a unix socket? is it not possible to open it up to a localhost http communication? because now, the way I access the slurm socket through the ui without running it with `sudo` is to `chmod 777 slurmrestd.socket`. before i did it, got an error saying Permission denied.
Thanks in advance:)

**Comment by rezib** [comment]
Hi @Talavig, I converted the issue into a Q&A discussion thread as it is arguably more a set of questions than an actual bug.

> Its important to mention that i only manage to reach slurmrestd by using sudo in my curl command. maybe that's related?
> Why is this error thrown?

The error your are facing could be explained by permission issue on slurmrestd Unix socket. In the [documented configuration](https://docs.rackslab.io/slurm-web/install/quickstart.html), we run both `slurmrestd` and Slurm-web agent with _slurm_ system user. This has to be defined with a drop-in configuration override for `slurmrestd` systemd service, this is [configured by default](https://github.com/rackslab/Slurm-web/blob/main/lib/systemd/slurm-web-agent.service#L8) in Slurm-web agent service. This way, Slurm-web agent has the permission to access `slurmrestd` Unix socket.

> In addition, can you only reach slurmrestd through a unix socket? is it not possible to open it up to a localhost http communication?

As stated in [architecture documentation](https://docs.rackslab.io/slurm-web/overview/architecture.html#protocols), Slurm-web only supports the Unix socket for the moment. It might change in the future with #313.

**Reply by Talavig** [reply]
When I try to run slurmrestd as user `slurm`, I get the following error:
```
Jul 16 11:10:27 standalone systemd[1]: slurmrestd.service: Failed with result 'exit-code'.
Jul 16 11:10:27 standalone systemd[1]: slurmrestd.service: Main process exited, code=exited, status=1/F>
Jul 16 11:10:27 standalone slurmrestd[2052]: slurmrestd: fatal: slurmrestd should not be run as SlurmUs>
Jul 16 11:10:27 standalone slurmrestd[2052]: fatal: slurmrestd should not be run as SlurmUser
Jul 16 11:10:27 standalone slurmrestd[2052]: debug:  main: server listen mode activated
Jul 16 11:10:27 standalone slurmrestd[2052]: debug:  NodeNames=localhost setting Sockets=4 based on CPU>
Jul 16 11:10:27 standalone slurmrestd[2052]: debug:  Reading slurm.conf file: /etc/slurm/slurm.conf
Jul 16 11:10:27 standalone slurmrestd[2052]: debug:  slurm_conf_init: using config_file=/etc/slurm/slur>
Jul 16 11:10:27 standalone slurmrestd[2052]: debug2: _establish_config_source: using config_file=/etc/s>
Jul 16 11:10:27 standalone systemd[1]: Started Slurm REST daemon.
```

This goes directly against what you have told me to do.
I don't know why this happens. Is there a workaround for this? maybe give the slurm-web user permissions to use the socket?
The following is my unit file for slurmrestd:
```
[Unit]
Description=Slurm REST daemon
After=network.target munge.service
ConditionPathExists=/etc/slurm/slurm.conf
[Service]
ExecStart=
Environment=
ExecStart=/usr/sbin/slurmrestd -vvv unix:/var/run/slurmrestd/slurmrestd.socket
RuntimeDirectory=slurmrestd
RuntimeDirectoryMode=0755
User=slurm
Group=slurm
```

**Reply by rezib** [reply]
What version of Slurm are you running?

Can you show me the output of these commands:
* `systemctl cat slurmrestd.service`
* `scontrol show config | grep SlurmUser`
* `getent passwd slurm`

Can you try to add `Environment=SLURMRESTD_SECURITY=disable_user_check` in your slurmrestd service override?

I have never met the error `fatal: slurmrestd should not be run as SlurmUser` in my testing environments, I would like to understand why.

**Reply by Talavig** [reply]
I have checked what version we run using `sinfo -V`, we are running `slurm-wlm 21.08.5`
here are all the commands you have asked for:
`$ systemctl cat slurmrestd.service`
```
# /lib/systemd/system/slurmrestd.service
[Unit]
Description=Slurm REST daemon
After=network-online.target munge.service slurmctld.service
Wants=network-online.target
ConditionPathExists=/etc/slurm/slurm.conf
Documentation=man:slurmrestd(8)

[Service]
Type=simple
EnvironmentFile=-/etc/default/slurmrestd
# Default to local auth via socket
ExecStart=/usr/sbin/slurmrestd $SLURMRESTD_OPTIONS unix:/run/slurmrestd.socket
# Uncomment to enable listening mode
#Environment="SLURM_JWT=daemon"
#ExecStart=/usr/sbin/slurmrestd $SLURMRESTD_OPTIONS 0.0.0.0:6820
ExecReload=/bin/kill -HUP $MAINPID

[Install]
WantedBy=multi-user.target

# /etc/systemd/system/slurmrestd.service.d/override.conf
[Unit]
Description=Slurm REST daemon
After=network.target munge.service
ConditionPathExists=/etc/slurm/slurm.conf


[Service]
#User=slurmrestuser
#Group=slurmrestgroup
#ExecStart=/usr/sbin/slurmrestd $SLURMRESTD_OPTIONS
#Environment=SLURMRESTD_LISTEN=:6820

ExecStart=
Environment=
ExecStart=/usr/sbin/slurmrestd -vvv unix:/var/run/slurmrestd/slurmrestd.socket
Environment=SLURMRESTD_SECURITY=disable_user_check
#ExecStart=/usr/sbin/slurmrestd -vvv localhost:8080 unix:/var/run/slurmrestd/slurmrestd.socket
#ExecStart=/usr/sbin/slurmrestd -vvv unix:/var/run/slurmrestd/slurmrestd.socket
#Environment=SLURMRESTD_LISTEN=localhost:6820,unix:/var/run/slurmrestd/slurmrestd.socket
#Environment=SLURMRESTD_LISTEN=:6820
#Environment=SLURM_JWT=daemon
RuntimeDirectory=slurmrestd
RuntimeDirectoryMode=0755
User=slurm
Group=slurm
```
`$ scontrol show config | grep SlurmUser`
```
SlurmUser               = slurm(64030)
```
`getent passwd slurm`
```
slurm:x:64030:64030::/nonexistent:/usr/sbin/nologin
```
In addition, I tried adding the env var you told me in the override, and I get the following error in the agent:
```
Thread-38 (process_request_thread): [INFO] 127.0.0.1 - - [16/Jul/2024 13:28:26] "GET /v3.1.0/stats HTTP/1.1" 500 -
Jul 16 13:28:26 standalone slurm-web-agent[2310]: KeyError: 'warnings'
Jul 16 13:28:26 standalone slurm-web-agent[2310]:     if len(result["warnings"]):
Jul 16 13:28:26 standalone slurm-web-agent[2310]:   File "/usr/lib/python3/dist-packages/slurmweb/views/agent.py", line 64, in slurmrest
Jul 16 13:28:26 standalone slurm-web-agent[2310]:     items = func(*args)
Jul 16 13:28:26 standalone slurm-web-agent[2310]:   File "/usr/lib/python3/dist-packages/slurmweb/views/agent.py", line 76, in filter_fields
Jul 16 13:28:26 standalone slurm-web-agent[2310]:     return func(*args)
Jul 16 13:28:26 standalone slurm-web-agent[2310]:   File "/usr/lib/python3/dist-packages/slurmweb/views/agent.py", line 88, in _cached_data
Jul 16 13:28:26 standalone slurm-web-agent[2310]:     return _cached_data(
Jul 16 13:28:26 standalone slurm-web-agent[2310]:   File "/usr/lib/python3/dist-packages/slurmweb/views/agent.py", line 101, in _cached_jobs
Jul 16 13:28:26 standalone slurm-web-agent[2310]:     for job in _cached_jobs():
Jul 16 13:28:26 standalone slurm-web-agent[2310]:   File "/usr/lib/python3/dist-packages/slurmweb/views/agent.py", line 236, in stats
Jul 16 13:28:26 standalone slurm-web-agent[2310]:     return view(*args, **kwargs)
Jul 16 13:28:26 standalone slurm-web-agent[2310]:   File "/usr/lib/python3/dist-packages/rfl/web/tokens.py", line 93, in wrapped
Jul 16 13:28:26 standalone slurm-web-agent[2310]:     return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
Jul 16 13:28:26 standalone slurm-web-agent[2310]:   File "/usr/lib/python3/dist-packages/flask/app.py", line 1499, in dispatch_request
Jul 16 13:28:26 standalone slurm-web-agent[2310]:     rv = self.dispatch_request()
Jul 16 13:28:26 standalone slurm-web-agent[2310]:   File "/usr/lib/python3/dist-packages/flask/app.py", line 1513, in full_dispatch_request
Jul 16 13:28:26 standalone slurm-web-agent[2310]:     rv = self.handle_user_exception(e)
Jul 16 13:28:26 standalone slurm-web-agent[2310]:   File "/usr/lib/python3/dist-packages/flask/app.py", line 1515, in full_dispatch_request
Jul 16 13:28:26 standalone slurm-web-agent[2310]:     response = self.full_dispatch_request()
Jul 16 13:28:26 standalone slurm-web-agent[2310]:   File "/usr/lib/python3/dist-packages/flask/app.py", line 2070, in wsgi_app
Jul 16 13:28:26 standalone slurm-web-agent[2310]: Traceback (most recent call last):
Jul 16 13:28:26 standalone slurm-web-agent[2310]: Thread-38 (process_request_thread): [ERROR] Exception on /v3.1.0/stats [GET]
```

**Reply by rezib** [reply, marked as answer]
> I have checked what version we run using `sinfo -V`, we are running `slurm-wlm 21.08.5` here are all the commands you have asked 

OK, unfortunately you are actually running a version of Slurm that is too old… As [stated in the quickstart guide](https://docs.rackslab.io/slurm-web/install/quickstart.html#_requirements), Slurm-web requires Slurm >= 23.02.

BTW, I suspect a regression in Slurm >= 22.05.7 introduced with commit https://github.com/SchedMD/slurm/commit/0a07c53c69a4dc468545998974e7e8a15a768c01 that may cause slurm user fatal error to never be triggered in more recent versions of Slurm. I will check this in more depth.

I will also try to do something in Slurm-web to report clearer error when the version of Slurm is not supported.

**Reply by rezib** [reply]
For the record, I have just opened #316, #319 and #320 following this discussion.

**Reply by Talavig** [reply]
After many agonizing hours, I finally managed to upgrade slurm (I'm new to slurm).
When running `sinfo -V`, I get that my version now is 23.11.8, which should work according to what you have told me.
I now try to access the ui and i get the following error:
```
Thread-196 (process_request_thread): [INFO] 127.0.0.1 - - [12/Aug/2024 13:18:53] "GET /v3.1.0/stats HTTP/1.1" 500 -
simplejson.errors.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
    return self.scan_once(s, idx=_w(s, idx).end())
  File "/usr/lib/python3/dist-packages/simplejson/decoder.py", line 400, in raw_decode
    obj, end = self.raw_decode(s)
  File "/usr/lib/python3/dist-packages/simplejson/decoder.py", line 370, in decode
    return _default_decoder.decode(s)
  File "/usr/lib/python3/dist-packages/simplejson/__init__.py", line 525, in loads
    return complexjson.loads(self.text, **kwargs)
  File "/usr/lib/python3/dist-packages/requests/models.py", line 900, in json
    result = response.json()
  File "/usr/lib/python3/dist-packages/slurmweb/views/agent.py", line 51, in slurmrest
    items = func(*args)
  File "/usr/lib/python3/dist-packages/slurmweb/views/agent.py", line 76, in filter_fields
    return func(*args)
  File "/usr/lib/python3/dist-packages/slurmweb/views/agent.py", line 88, in _cached_data
    return _cached_data(
  File "/usr/lib/python3/dist-packages/slurmweb/views/agent.py", line 101, in _cached_jobs
    for job in _cached_jobs():
  File "/usr/lib/python3/dist-packages/slurmweb/views/agent.py", line 236, in stats
    return view(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/rfl/web/tokens.py", line 93, in wrapped
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "/usr/lib/python3/dist-packages/flask/app.py", line 1499, in dispatch_request
    rv = self.dispatch_request()
  File "/usr/lib/python3/dist-packages/flask/app.py", line 1513, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/lib/python3/dist-packages/flask/app.py", line 1515, in full_dispatch_request
    response = self.full_dispatch_request()
  File "/usr/lib/python3/dist-packages/flask/app.py", line 2070, in wsgi_app
```
I tried running the commands you supplied to me earlier in our discussion, nothing seems to have been changed.
Do you have any idea why this may happen?

**Reply by rezib** [reply]
@Talavig can you give me the output of these commands:

* `systemctl cat slurmrestd.service`
* `curl --unix-socket /run/slurmrestd/slurmrestd.socket http://slurm/slurm/v0.0.39/jobs`

**Reply by Talavig** [reply]
`systemctl cat slurmrestd.service`
```
# /lib/systemd/system/slurmrestd.service
[Unit]
Description=Slurm REST daemon
After=network-online.target munge.service slurmctld.service
Wants=network-online.target
ConditionPathExists=/etc/slurm/slurm.conf
Documentation=man:slurmrestd(8)

[Service]
Type=simple
EnvironmentFile=-/etc/default/slurmrestd
# Default to local auth via socket
ExecStart=/usr/sbin/slurmrestd $SLURMRESTD_OPTIONS unix:/run/slurmrestd.socket
# Uncomment to enable listening mode
#Environment="SLURM_JWT=daemon"
#ExecStart=/usr/sbin/slurmrestd $SLURMRESTD_OPTIONS 0.0.0.0:6820
ExecReload=/bin/kill -HUP $MAINPID

[Install]
WantedBy=multi-user.target

# /etc/systemd/system/slurmrestd.service.d/override.conf
[Unit]
Description=Slurm REST daemon
After=network.target munge.service
ConditionPathExists=/etc/slurm/slurm.conf


[Service]
#User=slurmrestuser
#Group=slurmrestgroup
#ExecStart=/usr/sbin/slurmrestd $SLURMRESTD_OPTIONS
#Environment=SLURMRESTD_LISTEN=:6820

ExecStart=
Environment=
ExecStart=/usr/sbin/slurmrestd -vvv unix:/var/run/slurmrestd/slurmrestd.socket
Environment=SLURMRESTD_SECURITY=disable_user_check
#ExecStart=/usr/sbin/slurmrestd -vvv localhost:8080 unix:/var/run/slurmrestd/slurmrestd.socket
#ExecStart=/usr/sbin/slurmrestd -vvv unix:/var/run/slurmrestd/slurmrestd.socket
#Environment=SLURMRESTD_LISTEN=localhost:6820,unix:/var/run/slurmrestd/slurmrestd.socket
#Environment=SLURMRESTD_LISTEN=:6820
#Environment=SLURM_JWT=daemon
RuntimeDirectory=slurmrestd
RuntimeDirectoryMode=0755
User=slurm
Group=slurm
```
`curl --unix-socket /run/slurmrestd/slurmrestd.socket http://slurm/slurm/v0.0.39/jobs`
```
Authentication failure
```
I would like to mention that when running the curl command using the slurm user or simply sudo it works properly, and slurm agent runs with the slurm user (checked using `systemctl cat slurm-web-agent.service`).

**Reply by rezib** [reply]
Then what is the result of curl command run with slurm (or root) user?

It seems you are running `slurmrestd` with verbose logs, can you send the logs generated by this service when the slurm-web-agent stacktrace is generated?

**Reply by Talavig** [reply]
I have tried solving things on my own, it appears to be a problem with the db. here are the slurmctld logs right after a reset and a request sent via the ui:
```
Starting Slurm controller daemon...
slurmctld.service: Can't open PID file /run/slurmctld.pid (yet?) after start: Operation not permitted
Started Slurm controller daemon.
error: Configured MailProg is invalid
Job accounting information stored, but details not gathered
slurmctld version 24.05.2 started on cluster cluster
plugin_load_from_file: Incompatible Slurm plugin /usr/lib/slurm/select_cray_aries.so version (22.05.9)
error: Couldn't load specified plugin name for select/cray_aries: Incompatible plugin version
plugin_load_from_file: Incompatible Slurm plugin /usr/lib/slurm/select_cons_res.so version (22.05.9)
error: Couldn't load specified plugin name for select/cons_res: Incompatible plugin version
plugin_load_from_file: Incompatible Slurm plugin /usr/lib/slurm/mpi_pmix_v4.so version (22.05.9)
error: Couldn't load specified plugin name for mpi/pmix_v4: Incompatible plugin version
error: MPI: Cannot create context for mpi/pmix_v4
plugin_load_from_file: Incompatible Slurm plugin /usr/lib/slurm/mpi_pmix.so version (22.05.9)
error: Couldn't load specified plugin name for mpi/pmix: Incompatible plugin version
error: MPI: Cannot create context for mpi/pmix
plugin_load_from_file: Incompatible Slurm plugin /usr/lib/slurm/mpi_none.so version (22.05.9)
error: Couldn't load specified plugin name for mpi/none: Incompatible plugin version
error: MPI: Cannot create context for mpi/none
error: slurm_persist_conn_open: Failed to unpack persistent connection init resp message from localhost:3306
error: Sending PersistInit msg: Invalid argument
accounting_storage/slurmdbd: clusteracct_storage_p_register_ctld: Registering slurmctld at port 6817 with slurmdbd
error: slurm_persist_conn_open: Failed to unpack persistent connection init resp message from localhost:3306
error: Sending PersistInit msg: Invalid argument
error: slurm_persist_conn_open: Failed to unpack persistent connection init resp message from localhost:3306
error: Sending PersistInit msg: Invalid argument
No memory enforcing mechanism configured.
Recovered state of 1 nodes
Recovered information about 0 jobs
select/cons_tres: part_data_create_array: select/cons_tres: preparing for 1 partitions
Recovered state of 0 reservations
read_slurm_conf: backup_controller not specified
select/cons_tres: select_p_reconfigure: select/cons_tres: reconfigure
select/cons_tres: part_data_create_array: select/cons_tres: preparing for 1 partitions
Running as primary controller
error: slurm_persist_conn_open: Failed to unpack persistent connection init resp message from localhost:3306
error: Sending PersistInit msg: Invalid argument
```
The last two lines repeat indefinitely. 
here is my current slurmdbd.conf:
```
StorageType=accounting_storage/mysql
DbdAddr=localhost
DbdHost=localhost
StoragePort=3306
StorageHost=localhost
StorageLoc=slurm_acct_db
StorageUser=slurm
SlurmUser=slurm
StoragePass=********
LogFile=/var/log/slurmdbd.log
```
and here is my slurm.conf:
```
# slurm.conf file generated by configurator easy.html.
# Put this file on all nodes of your cluster.
# See the slurm.conf man page for more information.
#
SlurmctldHost=localhost
#
#MailProg=/bin/mail
MpiDefault=none
#MpiParams=ports=#-#
ProctrackType=proctrack/linuxproc
ReturnToService=1
SlurmctldPidFile=/var/run/slurmctld.pid
#SlurmctldPort=6817
SlurmdPidFile=/var/run/slurmd.pid
#SlurmdPort=6818
SlurmdSpoolDir=/var/spool/slurm/slurmd
SlurmUser=slurm
#SlurmdUser=root
StateSaveLocation=/var/spool/slurm
SwitchType=switch/none
TaskPlugin=task/affinity
#
#
# TIMERS
#KillWait=30
#MinJobAge=300
#SlurmctldTimeout=120
#SlurmdTimeout=300
#
#
# SCHEDULING
SchedulerType=sched/backfill
SelectType=select/cons_tres
SelectTypeParameters=CR_Core
#
#
# LOGGING AND ACCOUNTING
AccountingStorageType=accounting_storage/slurmdbd
AccountingStorageUser=slurm
AccountingStorageHost=localhost
AccountingStoragePort=3306

ClusterName=cluster
#JobAcctGatherFrequency=30
JobAcctGatherType=jobacct_gather/none
#SlurmctldDebug=info
#SlurmctldLogFile=
#SlurmdDebug=info
#SlurmdLogFile=
#
#
# COMPUTE NODES
NodeName=standalone State=idle Feature=dcv2,other
# NodeName=linux[1-32] CPUs=1 State=UNKNOWN
# NodeName=linux1 NodeAddr=128.197.115.158 CPUs=4 State=UNKNOWN
# NodeName=linux2 NodeAddr=128.197.115.7 CPUs=4 State=UNKNOWN

PartitionName=test Nodes=standalone Default=YES MaxTime=INFINITE State=UP
# PartitionName=test Nodes=standalone,linux[1-32] Default=YES MaxTime=INFINITE State=UP

# DefMemPerNode=1000
# MaxMemPerNode=1000
# DefMemPerCPU=4000
# MaxMemPerCPU=4096
```
slurmdbd logs seem fine.

**Reply by Talavig** [reply]
I have managed to get rid of the Invalid Argument messages, but still nothing works (it appears to have happened because of the port configuration).
I have noticed the following logs in the mariadb service:
```
Aug 13 12:31:45 standalone mariadbd[9385]: 2024-08-13 12:31:45 64 [Warning] Aborted connection 64 to db: 'unconnected' user: 'unauthenticated' host: 'localhost' (This connection closed normally without authentication)
```
These correlate with me trying to enter the ui. Is there a chance that for some reason, slurm does not try to use the user given in `StorageUser`? When trying to curl, I get a message saying the following:
```
"errors": [
    {
      "description": "openapi_get_db_conn() failed to open slurmdb connection",
      "error_number": 7000,
      "error": "Unable to connect to database",
      "source": "init_connection"
    }
  ],
```
But there is nothing outstaing in any of the logs: not in slurmdbd, slurmctld, slurmd and slurmrestd.

**Reply by rezib** [reply]
I normally don't do Slurm support on this channel but there is something that seems obvious here. The parameter `AccountingStoragePort` in `slurm.conf` must not be MySQL port but `slurmdbd` port instead (the default value is probably fine as well).

**Reply by Talavig** [reply]
To the best of my knowledge, I have now fixed all my db errors, and I still get the above mentioned error. What else can be the cause of this? do you nedd us to run other commads?

**Reply by rezib** [reply]
I am sorry but I can't figure out in the thread the above mentioned error you are referring to. What error are you still encountering?

**Reply by Talavig** [reply]
> After many agonizing hours, I finally managed to upgrade slurm (I'm new to slurm). When running `sinfo -V`, I get that my version now is 23.11.8, which should work according to what you have told me. I now try to access the ui and i get the following error:
> 
> ```
> Thread-196 (process_request_thread): [INFO] 127.0.0.1 - - [12/Aug/2024 13:18:53] "GET /v3.1.0/stats HTTP/1.1" 500 -
> simplejson.errors.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
>     return self.scan_once(s, idx=_w(s, idx).end())
>   File "/usr/lib/python3/dist-packages/simplejson/decoder.py", line 400, in raw_decode
>     obj, end = self.raw_decode(s)
>   File "/usr/lib/python3/dist-packages/simplejson/decoder.py", line 370, in decode
>     return _default_decoder.decode(s)
>   File "/usr/lib/python3/dist-packages/simplejson/__init__.py", line 525, in loads
>     return complexjson.loads(self.text, **kwargs)
>   File "/usr/lib/python3/dist-packages/requests/models.py", line 900, in json
>     result = response.json()
>   File "/usr/lib/python3/dist-packages/slurmweb/views/agent.py", line 51, in slurmrest
>     items = func(*args)
>   File "/usr/lib/python3/dist-packages/slurmweb/views/agent.py", line 76, in filter_fields
>     return func(*args)
>   File "/usr/lib/python3/dist-packages/slurmweb/views/agent.py", line 88, in _cached_data
>     return _cached_data(
>   File "/usr/lib/python3/dist-packages/slurmweb/views/agent.py", line 101, in _cached_jobs
>     for job in _cached_jobs():
>   File "/usr/lib/python3/dist-packages/slurmweb/views/agent.py", line 236, in stats
>     return view(*args, **kwargs)
>   File "/usr/lib/python3/dist-packages/rfl/web/tokens.py", line 93, in wrapped
>     return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
>   File "/usr/lib/python3/dist-packages/flask/app.py", line 1499, in dispatch_request
>     rv = self.dispatch_request()
>   File "/usr/lib/python3/dist-packages/flask/app.py", line 1513, in full_dispatch_request
>     rv = self.handle_user_exception(e)
>   File "/usr/lib/python3/dist-packages/flask/app.py", line 1515, in full_dispatch_request
>     response = self.full_dispatch_request()
>   File "/usr/lib/python3/dist-packages/flask/app.py", line 2070, in wsgi_app
> ```
> 
> I tried running the commands you supplied to me earlier in our discussion, nothing seems to have been changed. Do you have any idea why this may happen?

**Reply by rezib** [reply]
What is the output of `sudo curl --unix-socket /run/slurmrestd/slurmrestd.socket http://slurm/slurm/v0.0.39/jobs`?

**Reply by Talavig** [reply]
```
{
  "meta": {
    "plugin": {
      "type": "openapi\/v0.0.39",
      "name": "Slurm OpenAPI v0.0.39",
      "data_parser": "v0.0.39"
    },
    "client": {
      "source": "\/var\/run\/slurmrestd\/slurmrestd.socket->socket:[27969] (fd 8)"
    },
    "Slurm": {
      "version": {
        "major": 24,
        "micro": 2,
        "minor": 5
      },
      "release": "24.05.2"
    }
  },
  "errors": [
  ],
  "warnings": [
    {
      "description": "Zero jobs to dump"
    }
  ],
  "jobs": [
  ]
```

**Reply by Talavig** [reply]
Hi, are there any updates regarding this issue? From my curl output, there seems to be no error in my slurm installation, so what may be wrong here?

**Reply by rezib** [reply]
> Hi, are there any updates regarding this issue? From my curl output, there seems to be no error in my slurm installation, so what may be wrong here?

Hi @Talavig , I can't figure out how Slurm-web could fail with JSON decode error on `/stats` endpoint now that slurmrestd `/jobs` endpoint response is valid JSON, it makes no sense to me.

**Reply by Goranmk22** [reply]
Hello,

I had the same problem, can you try to modify these values in /etc/slurm/slurm.conf file

![image](https://github.com/user-attachments/assets/687d4333-18fb-403c-8760-15690c6ac7dc)

After these it worked for me.
